tfe/data_source_workspace_test.go

```diff
@@ -132,13 +162,35 @@ resource "tfe_workspace" "foobar" {
   terraform_version     = "0.11.1"
   trigger_prefixes      = ["/modules", "/shared"]
   working_directory     = "terraform/test"
-	global_remote_state   = true
+  global_remote_state   = true
 }
 
 data "tfe_workspace" "foobar" {
   name         = tfe_workspace.foobar.name
   organization = tfe_workspace.foobar.organization
-	depends_on   = [tfe_workspace.foobar]
+  depends_on   = [tfe_workspace.foobar]
+}`, rInt, rInt)
+}
+
+func testAccTFEWorkspaceDataSourceConfigWithTriggerPatterns(rInt int) string {
+	return fmt.Sprintf(`
+resource "tfe_organization" "foobar" {
+  name  = "tst-terraform-%d-ff-on"
+  email = "admin@company.com"
+}
+
+resource "tfe_workspace" "foobar" {
+  name                  = "workspace-test-%d"
+  organization          = tfe_organization.foobar.id
+  file_triggers_enabled = true
+  trigger_patterns      = ["/modules/**/*", "/**/network/*"]
+  working_directory     = "terraform/test"
+}
```

Comment:
Although this technically works as a data source test, it would be better to just **only** the data source. The way to do this is in the test itself `TestAccTFEWorkspaceDataSource_with_trigger_patterns`, to use the go-tfe client to create a workspace with the trigger patterns, then pass that workspace.name and org.name here in this `testAccTFEWorkspaceDataSourceConfigWithTriggerPatterns` config to be used as strings in the data source fields (name and organization). Then continue with verifying that the data is the same.

This way you also don't need a `depends_on`.